File: server/src/main/java/com/sentinel/server/user/dto/response/UserResponse.java
package com.sentinel.server.user.dto.response;

import com.sentinel.server.role.dto.response.RoleBriefResponse;
import com.sentinel.server.user.entity.UserStatus;
import java.time.Instant;
import java.util.List;

public record UserResponse(
        String id,
        String email,
        String displayName,
        UserStatus status,
        boolean tenantAdmin,
        List<RoleBriefResponse> roles,
        Instant createdAt,
        Instant updatedAt,
        Instant lastLoginAt) {}

File: frontend/src/features/users/dto/response/user.response.ts
export interface RoleBriefResponse {
  id: string;
  name: string;
}

export interface UserResponse {
  id: string;
  email: string;
  displayName: string;
  status: "ACTIVE" | "INACTIVE";
  tenantAdmin: boolean;
  roles: RoleBriefResponse[];
  createdAt: string;
  updatedAt: string;
  lastLoginAt: string | null;
}

export interface CreateUserResponse extends UserResponse {
  temporaryPassword: string;
}
